src/Microsoft.AspNetCore.Mvc.Analyzers/ViewFeaturesAnalyzerContext.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    public class ViewFeaturesAnalyzerContext
+    {
+#pragma warning disable RS1012 // Start action has no registered actions.
+        public ViewFeaturesAnalyzerContext(CompilationStartAnalysisContext context)
+#pragma warning restore RS1012 // Start action has no registered actions.
+        {
+            Context = context;
+        }
+
+        public CompilationStartAnalysisContext Context { get; }
+
+        private INamedTypeSymbol _htmlHelperType;
+        public INamedTypeSymbol HtmlHelperType => GetType(SymbolNames.IHtmlHelperType, ref _htmlHelperType);
```

Comment:
Instead of all the lazy caching lets just set this in the ctor. 1 dive wont hurt and its a lot more clear.